R/pkg/R/DataFrame.R

```diff
@@ -1191,6 +1191,9 @@ setMethod("collect",
                     vec <- do.call(c, col)
                     stopifnot(class(vec) != "list")
                     class(vec) <- PRIMITIVE_TYPES[[colType]]
+                    if (stringsAsFactors && is.character(vec)) {
```

Comment:
For performance maybe it is better to reverse the order of checks: `is.character(vec) && stringsAsFactors`

Reply:
Sure, thanks.